cpu/esp32/Kconfig

```diff
@@ -27,6 +27,8 @@ config CPU_FAM_ESP32
     select MODULE_ESP_IDF_DRIVER if TEST_KCONFIG
     select MODULE_ESP_IDF_ESP32 if TEST_KCONFIG
     select MODULE_ESP_IDF_SOC if TEST_KCONFIG
+    select MODULE_ZTIMER
+    select MODULE_ZTIMER_MSEC
```

Comment:
```suggestion
```
This shouldn't be necessary if correctly modelled. `ztimer_msec` should be required only if `esp_wifi_any`, `esp_eth` or `esp_i2c_hw` is used. Adding it by default shouldn't be the solution.

Reply:
I might be misunderstanding something but here I'm only reflecting on what is currently always built, `esp_ztimer.c` is in the main `esp32` directory therefore it's always built, so the module has a direct dependency to `ztimer_msec` independent of any other module. This might be wrong, but it out of the scope of what is being done in this PR.

Reply:
Yes, even though the file is compiled, the symbols are only required if `esp_wifi*` or `esp_eth` are used. So there is no necessity to add the module by default. Removing these dependencies from your PR works.

<details><summary><code>git diff</code></summary>

```diff
diff --git a/cpu/esp32/Kconfig b/cpu/esp32/Kconfig
index 74fb52122b..abad7755da 100644
--- a/cpu/esp32/Kconfig
+++ b/cpu/esp32/Kconfig
@@ -27,8 +27,6 @@ config CPU_FAM_ESP32
     select MODULE_ESP_IDF_DRIVER if TEST_KCONFIG
     select MODULE_ESP_IDF_ESP32 if TEST_KCONFIG
     select MODULE_ESP_IDF_SOC if TEST_KCONFIG
-    select MODULE_ZTIMER
-    select MODULE_ZTIMER_MSEC
     imply MODULE_NEWLIB_NANO
 
 ## CPU Models
diff --git a/cpu/esp32/Makefile.dep b/cpu/esp32/Makefile.dep
index 2ea6ef332e..3509e734c5 100644
--- a/cpu/esp32/Makefile.dep
+++ b/cpu/esp32/Makefile.dep
@@ -6,8 +6,6 @@ USEMODULE += esp_idf_driver
 USEMODULE += esp_idf_esp32
 USEMODULE += esp_idf_soc
 
-USEMODULE += ztimer_msec
-
 ifneq (,$(filter newlib,$(USEMODULE)))
   DEFAULT_MODULE += newlib_nano
 endif
diff --git a/cpu/esp_common/Makefile.dep b/cpu/esp_common/Makefile.dep
index f268d8eb13..553ea035e4 100644
--- a/cpu/esp_common/Makefile.dep
+++ b/cpu/esp_common/Makefile.dep
@@ -72,10 +72,6 @@ ifneq (,$(filter esp_freertos,$(USEMODULE)))
   USEMODULE += esp_freertos_common
 endif
 
-ifneq (,$(filter esp_freertos_common,$(USEMODULE)))
-  USEMODULE += ztimer_msec
-endif
-
 ifneq (,$(filter esp_now,$(USEMODULE)))
   USEMODULE += esp_wifi_any
 endif
diff --git a/cpu/esp_common/freertos/Kconfig b/cpu/esp_common/freertos/Kconfig
index e9d331b3e6..fabedae262 100644
--- a/cpu/esp_common/freertos/Kconfig
+++ b/cpu/esp_common/freertos/Kconfig
@@ -10,8 +10,6 @@ if TEST_KCONFIG && HAS_ARCH_ESP
 config MODULE_ESP_FREERTOS
     bool
     default y
-    select MODULE_ZTIMER
-    select MODULE_ZTIMER_MSEC
     help
       FreeRTOS to RIOT-OS adaption module for ESP SoCs source code compatibility.
 
```

</details>

 The compilation works without these dependencies. For example, [one of the the compilations](https://ci.riot-os.org/RIOT-OS/RIOT/17386/c706a6f5020af7b6504fb4644c385fe01b7a80d5/output/compile/tests/kconfig/esp32-olimex-evb:gnu.txt) that failed in last CI run works:

<details><summary><code>make BOARD=esp32-olimex-evb -C tests/kconfig</code></summary>

```
make: Verzeichnis „/home/gs/src/RIOT-Xtensa-ESP.working/tests/kconfig“ wird betreten
Building application "tests_kconfig" for "esp32-olimex-evb" with MCU "esp32".

"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/boards/esp32-olimex-evb
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/boards/common/esp32
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/core
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/cpu/esp32
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/cpu/esp32/freertos
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/cpu/esp32/periph
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/cpu/esp32/vendor
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/cpu/esp32/vendor/esp-idf
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/cpu/esp32/vendor/esp-idf/driver
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/cpu/esp32/vendor/esp-idf/esp32
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/cpu/esp32/vendor/esp-idf/soc
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/cpu/esp32/vendor/esp-idf/spi_flash
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/cpu/esp_common
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/cpu/esp_common/freertos
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/cpu/esp_common/periph
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/cpu/esp_common/vendor
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/cpu/esp_common/vendor/xtensa
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/drivers
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/drivers/periph_common
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/sys
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/sys/auto_init
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/sys/isrpipe
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/sys/log
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/sys/luid
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/sys/newlib_syscalls_default
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/sys/pm_layered
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/sys/random
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/sys/stdio_uart
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/sys/test_utils/interactive_sync
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/sys/tsrb
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/tests/kconfig/external_modules/external_module_1
"make" -C /home/gs/src/RIOT-Xtensa-ESP.working/tests/kconfig/external_modules/external_module_2
esptool.py v2.4.0
Parsing CSV input...
   text	   data	    bss	    dec	    hex	filename
  54986	   3648	   6356	  64990	   fdde	/home/gs/src/RIOT-Xtensa-ESP.working/tests/kconfig/bin/esp32-olimex-evb/tests_kconfig.elf
make: Verzeichnis „/home/gs/src/RIOT-Xtensa-ESP.working/tests/kconfig“ wird verlassen
```

</details>

Reply:
The fact that it build it justs because since the functions are normally unused the compiler does not detect that there is an undefined reference, but with the following I get a build error:

```diff
diff --git a/examples/hello-world/main.c b/examples/hello-world/main.c
index f51bf8c0a0..4f43ecbdfb 100644
--- a/examples/hello-world/main.c
+++ b/examples/hello-world/main.c
@@ -20,7 +20,14 @@
  */
 
 #include <stdio.h>
+#include "esp_common.h"
+#include "log.h"
 
+#include "esp_attr.h"
+#include "irq_arch.h"
+#include "xtimer.h"
+
+#include "rom/ets_sys.h"
 int main(void)
 {
     puts("Hello World!");
@@ -28,5 +35,8 @@ int main(void)
     printf("You are running RIOT on a(n) %s board.\n", RIOT_BOARD);
     printf("This board features a(n) %s MCU.\n", RIOT_MCU);
 
+    ETSTimer timer;
+    ets_timer_disarm(&timer);
+
     return 0;
 }
```

```
/data/riotbuild/riotbase/examples/hello-world/bin/esp32-wroom-32/cpu/esp_xtimer.o: In function `_ets_to_xtimer_get':
/data/riotbuild/riotbase/cpu/esp32/esp_xtimer.c:158: undefined reference to `xtimer_remove'
/data/riotbuild/riotbase/examples/hello-world/bin/esp32-wroom-32/cpu/esp_xtimer.o: In function `ets_timer_disarm':
/data/riotbuild/riotbase/cpu/esp32/esp_xtimer.c:165: undefined reference to `xtimer_remove'
```

Reply:
IMO this is a bug in the dependencies. How am I supposed to know when those functions might or might not be used? By getting a compiler error at some point. I'll exclude the source from the build then.

Reply:
Of course, but these functions are not intended to be used by the application. They are only for compatibility with the vendor libraries. Furthermore, the functions are not exposed and don't correspond to any RIOT API. So we don't have to take care that an application could use them.

ETS is the very old multitasking system in ESP-ROM, which in fact was never used. Even ESP-IDF rejects the support for these functions:

In `esp/esp-idf/components/esp32/ets_timer_legacy.c`
```c
/*
 * ets_timer module implements a set of legacy timer APIs which are
 * used by the WiFi driver. This is done on top of the newer esp_timer APIs.
 * Applications should not use ets_timer functions, as they may change without
 * notice.
 */
```
and further the documentation in `esp-idf/components/esp32/include/ets_sys.h`
```c
/**
  * @brief  Arm an ets timer, this timer range is 640 us to 429496 ms.
  *         In FreeRTOS, please call FreeRTOS apis, never call this api.
  * /
```